src/placement/jump_map.c

```diff
@@ -385,28 +274,18 @@ get_target(struct pool_domain *curr_dom, struct pool_target **target,
 				/* Get target id to check if target used */
 				dom_id = (*target)->ta_comp.co_id;
 
-				/*
-				 * Check to see if this target is valid to use.
-				 * You can reuse targets as long as there are
-				 * fewer targets than shards and all targets
-				 * have already been used
-				 */
-
-				for (i = 0; i < layout->ol_nr; ++i) {
-
-					tgt_id = layout->ol_shards[i].po_target;
-
-					if (tgt_id == dom_id)
-						break;
-				}
-			} while (i < shard_num);
+			} while (isset(tgts_used, dom_id));
```

Comment:
Will this run into a infinite loop if all domains are used? If the all domain are already used case isn't possible, it's better to add a D_ASSERT() before start this loop.

And should get_target() check the object type and relax the domain select criterion for striped object?

Reply:
I've pushed a patch to add a check before the loop to see if every target is used, and if so clear them.

While technically possible for it to just loop forever, that would only occur if an object is created with more shards than there are targets in the pool map. 
I'm not sure we would want to allow that.